### ISSUE TITLE
move signing to external wallet

### ISSUE THREAD

**Issue by DougAnderson444** [cryptidtech/multicid#2]
https://github.com/cryptidtech/multicid/blob/891a9bace3ec50e12717a917e7e2d0701915af16/src/vlad.rs#L186

This signing needs to be done from an external function.

**Comment by DougAnderson444** [comment]
I have this change compelted [here](https://github.com/DougAnderson444/multicid/tree/external-prover) but I will wait for https://github.com/cryptidtech/multikey/issues/3 and https://github.com/cryptidtech/multisig/pull/2 to resolve before opening the PR, and the dependency tree is downstream from those deps.

**Comment by DougAnderson444** [comment]
nudge @dhuseby

**Comment by dhuseby** [comment]
I just merged the multisig PR. Thanks

**Comment by dhuseby** [comment]
And poked Mike about the blsful update to crates.io

**Comment by dhuseby** [comment]
just fixed the multikey one

**Comment by dhuseby** [comment]
@DougAnderson444 I've been thinking for some time that there should be a standard protocol for activating external cryptographic services. Here's a proposal I made years ago for abstracting the cryptographic operations out of git: https://github.com/cryptidtech/git-cryptography-protocol/blob/main/Git%20Cryptography%20Protocol.md

It would be cool to generalize the proposal, update the spec, and implement it in Multikey/Multisig

**Comment by dhuseby** [comment]
FIDO is similar I think.

**Comment by DougAnderson444** [comment]
Yeah that's interesting. I think ultimately something like this is what is needed as part of the evolution.

I'll give it some thought. Implementing something like that would require an overhaul to the public API, likely having the user implement some signing trait that returns a future, instead of using a closure like it does now.
